astropy/nddata/mixins/ndarithmetic.py

```diff
@@ -24,211 +30,588 @@ class NDArithmeticMixin(object):
-            defined by the class used for the ``uncertainty`` attribute.
+        operation: `str`
+            The operation that is performed on the `NDData`. Supported are
+            ``addition``, ``subtraction``, ``multiplication`` and ``division``.
+
+        operand: `NDData` instance or something that can be converted to one.
+            The second operand in the operation.
+
+        uncertainty_correlation: ``Number`` or `~numpy.ndarray`
+            The correlation (rho) is defined between the uncertainties in
+            ``sigma_AB = sigma_A * sigma_B * rho`` (Latex?). If
+            ``handle_uncertainties`` is *not* ``True`` this will be ignored. A
+            value of ``0`` means uncorrelated (which is also the default)
+
+        meta_kwds_operate: ``None`` or `list`
+            If ``None`` no arithmetic meta operations are done. If this is a
+            list then each element of the list will be interpreted as a
+            keyword which should be arithmetically changed. In case both
+            operands have not-empty meta properties the resulting meta keyword
+            will be operand1 keyword operated upon operand2 keyword. If only
+            one operand has a not-empty meta, the resulting keyword will be
+            the keyword of the operand operated with the ``data`` (this is
+            only allowed if the data is actually a scalar) of the other
+            operand. (There will be more explanation and exampled sometime
+            soon). This parameter will be ignored if ``handle_meta`` is *not*
+            ``True``.
+
+        meta_kwds_set: ``None`` or `dict`
+            If ``None`` there are no keyword settings after the arithmetic
+            meta keyword operations. If this is a `dict` each key/value pair
+            in the ``meta_kwds_set`` will be added (or replaced if the keyword
+            already exists) in the results meta. This parameter will be ignored
+            if ``handle_meta`` is *not* ``True``.
+
+        handle_uncertainties: `bool` or ``None``
+            If ``None`` the result will have no uncertainty. If ``False`` the
+            result will have a copied version of the first operand (or the
+            second if the first had no uncertainty). In case this is ``True``
+            the result will have a correctly propagated uncertainty from the
+            uncertainties of the operands.
+
+        handle_mask: `bool` or ``None``
+            If ``None`` the result will have no mask. If ``False`` the result
+            will have a copied version of the mask of the first operand (or
+            if the first operand has no mask then the one from the second is
+            taken). If ``True`` the masks of both operands will be taken into
+            account and combined (by a bitwise ``or`` operation if the
+            :meth:`NDArithmeticMixin._arithmetic_mask` is not overriden).
+
+        handle_mata: `bool` or ``None``
```

Comment:
Renamed it, I'm currently working on the tests of NDArithmeticMixin and when they are done I will push the fix.